File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.cpp
#include <string>
#include <vector>

using namespace std;

void move(int disk, int now, int to_move, vector<vector<int>> &answer){
    if(disk == 0) return;
    
    move(disk - 1, now, 6 - now - to_move, answer);
    answer.push_back({now, to_move});
    move(disk - 1, 6 - now - to_move, to_move, answer);
}

vector<vector<int>> solution(int n) {
    vector<vector<int>> answer;
    move(n, 1, 3, answer);
    return answer;
}

File: 프로그래머스/2/12946. 하노이의 탑/하노이의 탑.py
def solution(n):
    answer = []
    def move(size, now, to_move):
        if size == 0: return
        
        move(size - 1, now, 6 - now - to_move)
        answer.append([now, to_move])
        move(size - 1, 6 - now - to_move, to_move)
    
    move(n, 1, 3)
    return answer